test/oli_web/live/remix_section_test.exs

```diff
@@ -90,6 +90,142 @@ defmodule OliWeb.RemixSectionLiveTest do
+
+      assert view |> element("#entry-#{unit1_container.revision.resource_id}") |> has_element?()
+      assert view |> element("#entry-#{revision1.resource_id}") |> has_element?()
+      assert view |> element("#entry-#{revision2.resource_id}") |> has_element?()
+    end
+
+    test "saving redirects instructor correctly", %{
+      conn: conn,
+      map: %{
+        section_1: section
+      }
+    } do
+      conn =
+        get(conn, Routes.live_path(OliWeb.Endpoint, OliWeb.Delivery.RemixSection, section.slug))
+
+      {:ok, view, _html} = live(conn)
+
+      render_hook(view, "reorder", %{"sourceIndex" => "0", "dropIndex" => "2"})
+
+      view
+      |> element("#save")
+      |> render_click()
+
+      assert_redirect(view, Routes.page_delivery_path(conn, :index, section.slug))
+    end
+
+    test "saving redirects open and free correctly", %{
+      conn: conn,
+      admin: admin,
+      map: %{
+        oaf_section_1: oaf_section_1
+      }
+    } do
+      conn =
+        Plug.Test.init_test_session(conn, %{})
+        |> Pow.Plug.assign_current_user(admin, OliWeb.Pow.PowHelpers.get_pow_config(:author))
+        |> get(
+          Routes.live_path(OliWeb.Endpoint, OliWeb.Delivery.RemixSection, oaf_section_1.slug)
+        )
+
+      {:ok, view, _html} = live(conn)
+
+      render_hook(view, "reorder", %{"sourceIndex" => "0", "dropIndex" => "2"})
+
+      view
+      |> element("#save")
+      |> render_click()
+
+      assert_redirect(view, OliWeb.OpenAndFreeView.get_path([:admin, :show, oaf_section_1]))
+    end
+
+    test "saving redirects product manager correctly", %{
+      conn: conn
+    } do
+      # create a product
+      %{
+        prod1: prod1,
+        author: product_author,
+        publication: publication,
+      } =
+        Seeder.base_project_with_resource2()
+        |> Seeder.create_product(%{title: "My 1st product", amount: Money.new(:USD, 100)}, :prod1)
+
+      {:ok, _prod} = Sections.create_section_resources(prod1, publication)
+
+      conn =
+        Plug.Test.init_test_session(conn, %{})
+        |> Pow.Plug.assign_current_user(
+          product_author,
+          OliWeb.Pow.PowHelpers.get_pow_config(:author)
+        )
+
+      conn =
+        get(conn, Routes.live_path(OliWeb.Endpoint, OliWeb.Delivery.RemixSection, prod1.slug))
+
+      {:ok, view, _html} = live(conn)
+
+      render_hook(view, "reorder", %{"sourceIndex" => "0", "dropIndex" => "2"})
+
+      view
+      |> element("#save")
+      |> render_click()
+
+      assert_redirect(
+        view,
+        Routes.live_path(OliWeb.Endpoint, OliWeb.Products.DetailsView, prod1.slug)
+      )
+    end
+
+    test "saving redirects admin correctly", %{
+      conn: conn,
+      map: %{
+        section_1: section_1,
+      },
+      admin: admin
+    } do
+      conn =
+        Plug.Test.init_test_session(conn, %{})
+        |> Pow.Plug.assign_current_user(admin, OliWeb.Pow.PowHelpers.get_pow_config(:author))
```

Comment:
NIT: thinking that we could split tests into different `describe` based on user type and make a common setup to avoid adding this to all admin tests in this case (not necessary to do the change here btw)

Reply:
Makes sense, wasn't super convinced with the way they were sharing setup either.